internal/service/target_apps/resource.go

```diff
@@ -0,0 +1,238 @@
+				Computed:            true,
+				MarkdownDescription: "The unique identifier of the target_app",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"description": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The description of the target_app",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+func (r *TargetAppResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*statsig.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *statsig.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Create builds a new target_app with the provided attributes.
+//
+// The ID of the created target_app is saved into the Terraform state once the value is returned from the API.
+// Statsig references objects by Name, which is unique. The ID is not used for identifying unique objects.
+func (r *TargetAppResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan TargetApp
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Map the Terraform plan data to the API request model
+	apiReq := statsig.TargetAppAPIRequest{
+		Name:        plan.Name.ValueString(),
+		Description: plan.Description.ValueString(),
+	}
+
+	// Create the target_app
+	target_app, err := r.client.CreateTargetApp(ctx, apiReq)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to create target_app, got error: %s", err),
+		)
+		return
+	}
+
+	// Update the plan attributes with the target_app attributes
+	plan = TargetApp{
+		Name:        types.StringValue(target_app.Name),
+		Description: types.StringValue(target_app.Description),
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("TargetApp created with Name: %s; and ID: %s", plan.Name, plan.ID))
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read fetches the target_app from the API and updates the Terraform state with the target_app attributes.
+func (r *TargetAppResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state TargetApp
+
+	// Get the current state of the resource
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the target_app from the API
+	target_app, err := r.client.GetTargetApp(ctx, state.ID.ValueString())
```

Comment:
Since statsig uses Names to identify resources, this should be
```suggestion
	target_app, err := r.client.GetTargetApp(ctx, state.Name.ValueString())
```